python_modules/dagster/dagster/_core/definitions/auto_materialize_sensor_definition.py

```diff
@@ -67,3 +152,43 @@ def asset_selection(self) -> AssetSelection:
     @property
     def sensor_type(self) -> SensorType:
         return SensorType.AUTO_MATERIALIZE
+
+
+class SchedulingConditionEvaluationSensorDefinition(SensorDefinition):
+    def __init__(
+        self,
+        name: str,
+        *,
+        asset_selection: CoercibleToAssetSelection,
+        run_tags: Optional[Mapping[str, Any]] = None,
+        default_status: DefaultSensorStatus = DefaultSensorStatus.STOPPED,
+        minimum_interval_seconds: Optional[int] = None,
+    ):
+        """Experimental variant of AutoMaterializeSensorDefinition that evaluates conditions in
+        user code.
+        """
+        self._run_tags = normalize_tags(run_tags).tags
+
+        super().__init__(
+            name=check_valid_name(name),
+            job_name=None,
+            evaluation_fn=evaluate_scheduling_conditions,
+            minimum_interval_seconds=minimum_interval_seconds,
+            default_status=default_status,
+            asset_selection=asset_selection,
+            # use a marker description to communicate capability to host process
+            description=EXECUTABLE_AMP_SENSOR_DESC,
+        )
+
+    @property
+    def run_tags(self) -> Mapping[str, str]:
+        return self._run_tags
+
+    @property
+    def asset_selection(self) -> AssetSelection:
+        return cast(AssetSelection, super().asset_selection)
+
+    @property
+    def sensor_type(self) -> SensorType:
+        # should likely be its own type
+        return SensorType.AUTO_MATERIALIZE
```

Comment:
I think we should add an additional type right away (immediate)/in this PR so we don't drive ourselves crazy.

Reply:
theres custom handling for `SensorType.AUTO_MATERIALIZE` in several places 

https://github.com/search?q=repo%3Adagster-io%2Fdagster%20SensorType.AUTO_MATERIALIZE&type=code

how far should I take this PR in making this sensor behave like that one?

Reply:
I guess as far as it takes to have it not be misleading? Given that these sensors act very differently I think it is important for them to have different types and be explicit where we are allowing one versus the other versus both of them.